lib/commands/appstore-upload.ts

```diff
@@ -0,0 +1,82 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import {StringCommandParameter} from "../common/command-params";
+import * as temp from "temp";
+import * as path from "path";
+
+export class PublishIOS implements ICommand {
+	constructor(private $errors: IErrors,
+		private $fs: IFileSystem,
+		private $hostInfo: IHostInfo,
+		private $injector: IInjector,
+		private $itmsTransporterService: IITMSTransporterService,
+		private $logger: ILogger,
+		private $options: IOptions,
+		private $prompter: IPrompter,
+		private $stringParameterBuilder: IStringParameterBuilder) { }
+
+	public allowedParameters: ICommandParameter[] = [new StringCommandParameter(this.$injector), new StringCommandParameter(this.$injector),
+			new StringCommandParameter(this.$injector), new StringCommandParameter(this.$injector)];
+
+	public execute(args: string[]): IFuture<void> {
+		return (() => {
+			let username = args[0],
+				password = args[1],
+				mobileProvisionIdentifier = args[2],
+				codeSignIdentity = args[3],
+				ipaFilePath = this.$options.ipa ? path.resolve(this.$options.ipa) : null,
+				bundleId = this.$itmsTransporterService.getBundleIdentifier(ipaFilePath).wait();
+
+			if(!username) {
+				username = this.$prompter.getString("Apple ID", { allowEmpty: false }).wait();
+			}
+
+			if(!password) {
+				password = this.$prompter.getPassword("Apple ID password").wait();
+			}
+
+			let iOSApplications = this.$itmsTransporterService.getiOSApplications({username, password}).wait();
+			if (!iOSApplications || !iOSApplications.length) {
+				this.$errors.failWithoutHelp("You don't have any applications registered at iTunes Connect.");
+			}
+
+			let iosApplication = _.find(iOSApplications, app => app.bundleId === bundleId);
+
+			if (!iosApplication) {
+				this.$errors.failWithoutHelp(`No application found on iTunes Connect that matches identifier ${bundleId}`);
+			}
+
+			if(!mobileProvisionIdentifier && !ipaFilePath) {
+				this.$logger.warn("Mobile Provision identifier not set - a default one will be used. You can set one in app/App_Resources/iOS/build.xcconfig");
```

Comment:
Mobile Provision identifier not set - a default one will be used. You can set one in app/App_Resources/iOS/build.xcconfig -> No mobile provision identifier set. A default mobile provision will be used. You can ....